configure.ac

```diff
@@ -64,6 +77,18 @@ if [[ -f "$header_prefix/include/libzfs/libzfs.h" ]] || [[ "$zof" = true ]]; the
 "
 	LDFLAGS="-L${header_prefix}/lib ${LDFLAGS}"
 
+elif [[ "$build_macos" = "yes" ]] && [[ "x$with_o3x" != "x"  ]]; then
+	zof=true
+
+	CFLAGS="${CFLAGS} 
+-I${with_o3x}/include/
+-I${with_o3x}/lib/libspl/include/
+-I./include/
+-D_MACHINE_ENDIAN_H_ -DHAVE_ISSETUGID
+-include ${with_o3x}/include/sys/zfs_mount.h
+"
+	LDFLAGS="-L${header_prefix}/lib ${LDFLAGS}"
+	AC_SUBST(HAVE_O3X, 1)
```

Comment:
`HAVE_O3X` won't be set as `0/False` when this is executed which will result in FreeBSD/linux builds failing. We need to fix this.

Reply:
We need to make sure in case it's not macos we are building this on, this is set to 0 ( like we have done for various cases below )

Reply:
This is now fixed in 9657ab4